homeassistant/components/mqtt/lock.py

```diff
@@ -148,8 +157,24 @@ def message_received(msg: ReceiveMessage) -> None:
             payload = self._value_template(msg.payload)
             if payload == self._config[CONF_STATE_LOCKED]:
                 self._attr_is_locked = True
+                self._attr_is_locking = False
+                self._attr_is_unlocking = False
+                self._attr_is_jammed = False
             elif payload == self._config[CONF_STATE_UNLOCKED]:
                 self._attr_is_locked = False
+                self._attr_is_locking = False
+                self._attr_is_unlocking = False
+                self._attr_is_jammed = False
+            elif payload == self._config[CONF_STATE_LOCKING]:
+                self._attr_is_locking = True
+                self._attr_is_unlocking = False
+                self._attr_is_jammed = False
+            elif payload == self._config[CONF_STATE_UNLOCKING]:
+                self._attr_is_locking = False
+                self._attr_is_unlocking = True
+                self._attr_is_jammed = False
+            elif payload == self._config[CONF_STATE_JAMMED]:
```

Comment:
May be the jammed condition should be handles different from the other states as it could happen when opening, closing or in open or closed state. That would require another state topic (and template) to listen to a payload for determining a jammed condition.

Reply:
OK, but how to determine the lock is no longer jammed?
With two payloads for jammed and not jammed respectively?
Because the example I know publish the jammed state to the same topic as state and has no other payload to specify the motor is no longer jammed.

Reply:
You need to treat jammed different because the other properties remain, only the state changes.
You will need two states/payloads to control this and a separate topic.